src/Selenium2Library/keywords/_element.py

```diff
@@ -602,41 +602,54 @@ def page_should_not_contain_image(self, locator, message='', loglevel='INFO'):
     def get_matching_xpath_count(self, xpath):
         """Returns number of elements matching `xpath`
 
-        One should not use the xpath= prefix for 'xpath'. XPath is assumed.
+        As of release 1.8 `Get Matching Xpath Count` allows the
+        argument `xpath` to start with the xpath= locator strategy.
 
-        Correct:
+        Example:
         | count = | Get Matching Xpath Count | //div[@id='sales-pop']
-        Incorrect:
         | count = | Get Matching Xpath Count | xpath=//div[@id='sales-pop']
 
         If you wish to assert the number of matching elements, use
         `Xpath Should Match X Times`.
         """
-        count = len(self._element_find("xpath=" + xpath, False, False))
-        return str(count)
+        if xpath.startswith('//'):
+            xpath = "xpath=" + xpath
+        
+        if xpath.startswith('xpath='):
```

Comment:
The detection of the xpath locator prefix is both case- and space-sensitive. This is not consistent with S2L.